xds/internal/balancer/clusterresolver/resource_resolver.go

```diff
@@ -280,16 +299,20 @@ func (rr *resourceResolver) generateLocked() {
 		}
 	}
 	select {
-	case <-rr.updateChannel:
+	case ru := <-rr.updateChannel:
+		if ru.onDone != nil {
+			ru.onDone.OnDone()
+		}
```

Comment:
Add comment here about how this is an optimization/opportunistic drain and thus need to invoke previous unprocessed updates onDone.

Reply:
Done.